trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingTrinoAwsProxyServer.java

```diff
@@ -210,6 +211,11 @@ public Builder withOpaContainer()
 
         public TestingTrinoAwsProxyServer buildAndStart()
         {
+            addModule(s3RequestRewriterModule("testing", TestingS3RequestRewriter.class, binder -> {
```

Comment:
I like this approach, my only concern is that we don't test the default NOOP `S3RequestRewriter` (not the testing NOOP).
It's true we also don't test the default `CredentialsProvider` or `AssumedRoleProvider` but for those an implementation is required for the aws-proxy to do anything useful.
For the `S3RequestRewriter` the noop will likely get a lot more use to its good to test it.

Adding my proposed changes as a path since it may be easier to make sense of:

```patch
Subject: [PATCH] rewrite testing
---
Index: trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/harness/TrinoAwsProxyTestExtension.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/harness/TrinoAwsProxyTestExtension.java b/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/harness/TrinoAwsProxyTestExtension.java
--- a/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/harness/TrinoAwsProxyTestExtension.java	(revision 7a173c987fb87fc09ce68b4523667b8d220cff6f)
+++ b/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/harness/TrinoAwsProxyTestExtension.java	(date 1724153777142)
@@ -19,6 +19,8 @@
 import io.trino.aws.proxy.server.remote.RemoteS3Facade;
 import io.trino.aws.proxy.server.testing.ContainerS3Facade;
 import io.trino.aws.proxy.server.testing.TestingS3ClientModule;
+import io.trino.aws.proxy.server.testing.TestingS3RequestRewriteController;
+import io.trino.aws.proxy.server.testing.TestingS3RequestRewriter;
 import io.trino.aws.proxy.server.testing.TestingTrinoAwsProxyServer;
 import io.trino.aws.proxy.server.testing.TestingUtil.ForTesting;
 import io.trino.aws.proxy.server.testing.containers.S3Container;
@@ -66,6 +68,8 @@
                             .setDefault()
                             .to(ContainerS3Facade.PathStyleContainerS3Facade.class)
                             .asEagerSingleton();
+                    newOptionalBinder(binder, TestingS3RequestRewriter.class).setDefault().toInstance(TestingS3RequestRewriter.NOP);
+                    binder.bind(TestingS3RequestRewriteController.class).in(Scopes.SINGLETON);
                 })
                 .buildAndStart();
         testingServersRegistry.put(extensionContext.getUniqueId(), trinoS3ProxyServer);
Index: trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/RequestRewriteUtil.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/RequestRewriteUtil.java b/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/RequestRewriteUtil.java
--- a/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/RequestRewriteUtil.java	(revision 7a173c987fb87fc09ce68b4523667b8d220cff6f)
+++ b/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/RequestRewriteUtil.java	(date 1724153784639)
@@ -27,6 +27,7 @@
 import java.util.UUID;
 
 import static com.google.inject.multibindings.OptionalBinder.newOptionalBinder;
+import static io.trino.aws.proxy.spi.plugin.TrinoAwsProxyServerBinding.s3RequestRewriterModule;
 
 public final class RequestRewriteUtil
 {
@@ -59,10 +60,11 @@
         {
             return builder
                     .withProperty("aws.proxy.s3.path", "/api/some/s3/path")
-                    .addModule(binder -> {
+                    .addModule(s3RequestRewriterModule("testing", TestingS3RequestRewriter.class, binder -> {
                         newOptionalBinder(binder, TestingS3RequestRewriter.class).setBinding().to(Rewriter.class).in(Scopes.SINGLETON);
                         binder.bind(SetupRequestRewrites.class).asEagerSingleton();
-                    });
+                    }))
+                    .withProperty("s3-request-rewriter.type", "testing");
         }
     }
 
Index: trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingTrinoAwsProxyServer.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingTrinoAwsProxyServer.java b/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingTrinoAwsProxyServer.java
--- a/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingTrinoAwsProxyServer.java	(revision 7a173c987fb87fc09ce68b4523667b8d220cff6f)
+++ b/trino-aws-proxy/src/test/java/io/trino/aws/proxy/server/testing/TestingTrinoAwsProxyServer.java	(date 1724153750004)
@@ -53,7 +53,6 @@
 import static io.trino.aws.proxy.server.testing.TestingUtil.TESTING_CREDENTIALS;
 import static io.trino.aws.proxy.spi.plugin.TrinoAwsProxyServerBinding.assumedRoleProviderModule;
 import static io.trino.aws.proxy.spi.plugin.TrinoAwsProxyServerBinding.credentialsProviderModule;
-import static io.trino.aws.proxy.spi.plugin.TrinoAwsProxyServerBinding.s3RequestRewriterModule;
 
 public final class TestingTrinoAwsProxyServer
         implements Closeable
@@ -211,11 +210,7 @@
 
         public TestingTrinoAwsProxyServer buildAndStart()
         {
-            addModule(s3RequestRewriterModule("testing", TestingS3RequestRewriter.class, binder -> {
-                binder.bind(TestingS3RequestRewriteController.class).in(Scopes.SINGLETON);
-                newOptionalBinder(binder, TestingS3RequestRewriter.class).setDefault().toInstance(TestingS3RequestRewriter.NOP);
-            }));
-            withProperty("s3-request-rewriter.type", "testing");
+
             if (addTestingCredentialsRoleProviders) {
                 if (mockS3ContainerAdded) {
                     modules.add(binder -> binder.bind(TestingCredentialsInitializer.class).asEagerSingleton());
```